pkg/declarativeconfig/auth_provider.go

```diff
@@ -0,0 +1,49 @@
+package declarativeconfig
+
+type RequiredAttribute struct {
+	AttributeKey   string `yaml:"key,omitempty"`
+	AttributeValue string `yaml:"value,omitempty"`
+}
+
+type ClaimMapping struct {
+	PathToClaim    string `yaml:"pathToClaim,omitempty"`
+	MapToClaimName string `yaml:"mapTo,omitempty"`
+}
+
+type OIDCConfig struct {
+	Issuer                    string `yaml:"issuer,omitempty"`
+	CallbackMode              string `yaml:"mode,omitempty"`
```

Comment:
I agree that guidance should be provided. I wonder if comments written in these files can be compiled into documentation, or should we introduce separate documentation file for that?

Reply:
I think for now it would be best to include the comments here.
I'm not sure how we are going to do the whole documentation for these configs later on in the actual documentation (my current idea would be to propose something similar like [dex documentation](https://dexidp.io/docs/connectors/github/), which I really like), but at the very least we can use the code comments as one of the sources.